client/components/infinite-list/index.jsx

```diff
@@ -161,6 +161,27 @@ module.exports = React.createClass( {
 		}
 	},
 
+	reset: function() {
```

Comment:
I would love to rip out a lot of infinite-scroll logic. I think we merged it (or rather I merged it) without really quantifying the benefit.  But the increased complexity is daunting.